test/lit/passes/O4_disable-bulk-memory.wast

```diff
@@ -80,8 +80,8 @@
  (global $global$6 i32 (i32.const 100))
  ;; CHECK:      (elem (i32.const 0) $null)
 
- ;; CHECK:      (export "memory" (memory $0))
- (export "memory" (memory $0))
+ ;; CHECK:      (export "memory" (memory $1))
+ (export "memory" (memory $1))
```

Comment:
This is because of what occurs on line 57: `(import "env" "memory" (memory $1 1))`. We import a memory with the name $1. Before multi-memories, wasm-validator would verify the memory export by checking whether the name was "0" (as it is in this test) or the name of the single memory (which would have failed as the memory is named $1) [(link)](https://sourcegraph.com/github.com/WebAssembly/binaryen@main/-/blob/src/wasm/wasm-validator.cpp?L2943&subtree=true). 

In a multi-memories world, wasm-validator now looks for the exported memory name (like for function/global/table exports) in our list of memories. I changed the name of the export in this test to actually match the name of the import.

Reply:
I see, thanks!